src/Core/Command/DeadLetterRetryCommand.php

```diff
@@ -0,0 +1,243 @@
+<?php declare(strict_types=1);
+/**
+ * This File is part of JTL-Software
+ *
+ * User: marius
+ * Date: 7/23/21
+ */
+
+namespace JTL\SCX\Lib\Channel\Core\Command;
+
+use DateTimeImmutable;
+use JTL\Nachricht\Contract\Serializer\MessageSerializer;
+use JTL\Nachricht\Contract\Transport\Amqp\AmqpQueueLister;
+use JTL\Nachricht\Serializer\Exception\DeserializationFailedException;
+use JTL\Nachricht\Transport\Amqp\AmqpHttpConnectionFailedException;
+use JTL\Nachricht\Transport\Amqp\AmqpTransport;
+use JTL\SCX\Lib\Channel\Contract\Core\Log\ScxLogger;
+use JTL\SCX\Lib\Channel\Event\AbstractEvent;
+use PhpAmqpLib\Message\AMQPMessage;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DeadLetterRetryCommand extends AbstractCommand
+{
+    protected static $defaultName = 'chn:dead-letter:retry';
+
+    private AmqpTransport $transport;
+    private AmqpQueueLister $queueLister;
+    private MessageSerializer $messageSerializer;
+    private $olderThan;
```

Comment:
type hints fehlen

generell würde ich diese filter und settings nicht als properties sehen, sondern lediglich den funktionen übergeben welche mit den Einstellungen arbeiten müssen.

Reply:
Beipiel

`private function processQueue(string $queue, bool $interactive): void`

hier machst du es in teilen ja bereits mit $interactive

Vorschlag:
`private function processQueue(string $queue, bool $interactive, $olderThan, $filterBylastError, $resetReceives): void`